src/Illuminate/Broadcasting/Broadcasters/AblyBroadcaster.php

```diff
@@ -139,97 +188,145 @@ public function broadcast(array $channels, $event, array $payload = [])
     }
 
     /**
-     * Build an Ably message object for broadcasting.
-     *
-     * @param  string  $event
-     * @param  array  $payload
-     * @return \Ably\Models\Message
+     * @param $channelName string
+     * @param $token string
+     * @param $clientId string
+     * @param string[] $channelCapability
+     * @return string
      */
-    protected function buildAblyMessage($event, array $payload = [])
+    function getSignedToken($channelName, $token, $clientId, $channelCapability = ['*'])
     {
-        return tap(new AblyMessage, function ($message) use ($event, $payload) {
-            $message->name = $event;
-            $message->data = $payload;
-            $message->connectionKey = data_get($payload, 'socket');
-        });
+        $header = array(
+            "typ" => "JWT",
+            "alg" => "HS256",
+            "kid" => $this->getPublicToken()
+        );
+        // Set capabilities for public channel as per https://ably.com/docs/core-features/authentication#capability-operations
+        $channelClaims = $this->defaultChannelClaims;
+        $serverTimeFn = function () {
```

Comment:
Why do we need to wrap this in a function? Would it not work to just call `self::getServerTime` directly when needed?

Reply:
Yeah, we can do that but it's to avoid getting `serverTime` earlier. When we pass the `serverTimeFn`, `time` is fetched at the time of validation. So, there is more accuracy while validating the token.

Reply:
Yeah I understand that, I was more just confused as to why we're wrapping it in a function when it's already a method. It's not important anyway so I'll resolve this conversation.